### PR TITLE
Cleanup tests/e2e/common directory - part 1

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/98326

This is supposed to be no-op.

```release-note
NONE
```

/kind cleanup
/priority important-soon

### CODE REVIEW

**Comment by wojtek-t** [comment]
@spiffxp - I don't know why after moving files to subdirectories they are no longer treated as conformance - can you please help?

**Comment by wojtek-t** [comment]
@spiffxp - I figured that out, it's ready for review

File: test/e2e_node/e2e_node_suite_test.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ import (
 	"github.com/onsi/gomega"
 	"github.com/spf13/pflag"
 	"k8s.io/klog/v2"
+
+	// test sources
+	_ "k8s.io/kubernetes/test/e2e_node/gcp"
```
Comment (spiffxp) [inline test/e2e_node/e2e_node_suite_test.go:59]: ?

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

File: test/e2e_node/e2e_node_suite_test.go (hunk 1 above)
Comment (spiffxp) [reply test/e2e_node/e2e_node_suite_test.go:59]: I gather this should have been here before and wasn't, don't see a move to this dir as part of this PR

**Comment by spiffxp** [comment]
/hold
actually, wait... I'm confused how this is picking up tests in subdirs of common

File: test/e2e_node/e2e_node_suite_test.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e_node/e2e_node_suite_test.go:59]: Yes - https://github.com/kubernetes/kubernetes/pull/99319 effectively broke it.

**Comment by wojtek-t** [comment]
> actually, wait... I'm confused how this is picking up tests in subdirs of common

https://github.com/kubernetes/kubernetes/pull/99374/files#diff-9f9944ba2dc44cdca30753ad296be4ee7e5dd85e801f9910cf056946859ce4f1R28-R30

@spiffxp - is that what you're asking for?

File: test/e2e/BUILD [hunk 2]
```diff
@@ -25,6 +25,9 @@ go_test(
         "//test/e2e/autoscaling:go_default_library",
         "//test/e2e/cloud:go_default_library",
         "//test/e2e/common:go_default_library",
+        "//test/e2e/common/network:go_default_library",
+        "//test/e2e/common/node:go_default_library",
+        "//test/e2e/common/storage:go_default_library",
```
Comment (spiffxp) [inline test/e2e/BUILD:30]: Add `test/e2e/common/imports.go` that does this, and drop these imports

This way you don't need to make the same change in `e2e_node` (it no longer sees tests in the subpackages)

File: test/e2e/e2e_test.go [hunk 3]
```diff
@@ -46,6 +46,9 @@ import (
 	_ "k8s.io/kubernetes/test/e2e/autoscaling"
 	_ "k8s.io/kubernetes/test/e2e/cloud"
 	_ "k8s.io/kubernetes/test/e2e/common"
+	_ "k8s.io/kubernetes/test/e2e/common/network"
+	_ "k8s.io/kubernetes/test/e2e/common/node"
+	_ "k8s.io/kubernetes/test/e2e/common/storage"
```
Comment (spiffxp) [inline test/e2e/e2e_test.go:51]: Add `test/e2e/common/imports.go` that does this, and drop these imports

This way you don't need to make the same change in `e2e_node` (it no longer sees tests in the subpackages)

File: test/e2e_node/e2e_node_suite_test.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e_node/e2e_node_suite_test.go:59]: I replaced with the imports pattern here too.

File: test/e2e_node/imports.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	// ensure these packages are scanned by ginkgo for e2e tests
+	_ "k8s.io/kubernetes/test/e2e_node/gcp"
```
Comment (spiffxp) [inline test/e2e_node/imports.go:21]: this may need to be directly in e2e_node_suite_test.go

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
going to wait for pull-kubernetes-node-e2e to confirm no delta in test results (except the e2e_node/gcp test) before removing /hold

**Comment by wojtek-t** [comment]
@spiffxp - I split the second commit to https://github.com/kubernetes/kubernetes/pull/99420 (to not block this PR on that)

**Review by spiffxp — APPROVED** [review]
/lgtm
/hold cancel

**Pull request merged**